packages/liveblocks-react-comments/src/slate/plugins/auto-links.ts

```diff
@@ -0,0 +1,316 @@
+  const [node, path] = entry;
+
+  // Step 1: Ensure that the Link node only contains text nodes as children
+  const children = Node.children(editor, path);
+  for (const [child] of children) {
+    if (Text.isText(child)) continue;
+    Transforms.unwrapNodes(editor, { at: path });
+    return;
+  }
+  // Attempt to match the text content (of the Link node) against the URL regex
+  const text = Node.string(node);
+  const match = URL_REGEX.exec(text);
+
+  // Step 2: Ensure that the text content of the Link node matches the URL regex and is identical to the match
+  if (!match || match[0] !== text) {
+    Transforms.unwrapNodes(editor, { at: path });
+    return;
+  }
+
+  // Step 3: Ensure that if the text content of the Link node ends with a period, we unwrap the Link node and wrap the text before the period in a new Link node
+  if (endsWithPeriod(text)) {
+    Transforms.unwrapNodes(editor, { at: path });
+
+    const textBeforePeriod = text.slice(0, text.length - 1);
+
+    // Remove the last character from the link text and wrap the remaining text in a new link node
+    Transforms.wrapNodes(
+      editor,
+      {
+        type: "auto-link",
+        href: textBeforePeriod,
+        children: [],
+      },
+      {
+        at: {
+          anchor: { path, offset: 0 },
+          focus: { path, offset: textBeforePeriod.length },
+        },
+        split: true,
+      }
+    );
+    return;
+  }
+
+  // Step 4: Ensure that the text content of the Link node is surrounded by separators or the start/end of the text content
+  if (!isPreviousNodeValid(editor, path) || !isNextNodeValid(editor, path)) {
+    Transforms.unwrapNodes(editor, { at: path });
+    return;
+  }
+
+  // Step 5: Ensure that the href attribute of the Link node is identical to its text content
+  if (node.href !== text) {
+    Transforms.setNodes(editor, { href: match[0] }, { at: path });
+    return;
+  }
+};
+
+const handleLinkCreate = (editor: Editor, entry: NodeEntry<Text>) => {
+  const [node, path] = entry;
+
+  // Step 1: Ensure that the text content of the node matches the URL regex
+  const match = URL_REGEX.exec(node.text);
+  if (!match) return;
+
+  const start = match.index;
+  const end = start + match[0].length;
+
+  // Step 2: Ensure that the content around the node is valid
+  if (!isContentAroundValid(editor, entry, start, end)) return;
+
+  Transforms.wrapNodes<ComposerBodyAutoLink>(
+    editor,
+    {
+      type: "auto-link",
+      href: match[0],
+      children: [],
+    },
+    {
+      at: {
+        anchor: { path, offset: start },
+        focus: { path, offset: end },
+      },
+      split: true,
+    }
+  );
+  return;
+};
+
+const handleNeighbours = (editor: Editor, entry: NodeEntry<Text>) => {
+  const [node, path] = entry;
+  const text = node.text;
+
+  const previousSibling = Editor.previous(editor, { at: path });
+
+  if (previousSibling && isComposerBodyAutoLink(previousSibling[0])) {
+    if (/^\.[a-zA-Z0-9]+/.test(text)) {
```

Comment:
We should probably define this regex once with the other ones (`URL_REGEX` and  `PUNCTUATION_OR_SPACE`) above.

Reply:
Good idea! Done!